main.go

```diff
@@ -1,48 +1,55 @@
 package main
 
 import (
-	"encoding/json"
+	"context"
 	"flag"
 	"log"
-	"net/http"
-	"strings"
+
+	"github.com/Clever/resolve-ip/gen-go/models"
+	"github.com/Clever/resolve-ip/gen-go/server"
+	"gopkg.in/Clever/kayvee-go.v5/logger"
 )
 
+var addr = flag.String("addr", ":8080", "Address to listen at")
 var path = flag.String("path", "./GeoLiteCity", "path to Geo Lite City db")
-var port = flag.String("port", "80", "port to listen on")
+
+type handler struct {
+	db *GeoDB
+}
+
+func (h handler) HealthCheck(ctx context.Context) error {
+	return nil
+}
+
+func (h handler) LocationForIP(ctx context.Context, i *models.LocationForIPInput) (*models.IP, error) {
+	latlon, err := h.db.Lookup(i.Ip)
+	if err == ErrIPMissing {
+		return nil, models.LocationForIP404Output{}
+	}
+	if err == ErrBadIP {
+		return nil, models.DefaultBadRequest{Msg: err.Error()}
+	}
+	if err != nil {
+		return nil, models.DefaultInternalError{Msg: err.Error()}
+	}
```

Comment:
originally I was going to do that. I had

```
400:
          description: IP is not a valid format
```
as one of the responses. Running `make generate` gave me

```
2016/10/12 03:54:35 building a plan for generation
2016/10/12 03:54:35 planning definitions
2016/10/12 03:54:35 planning operations
2016/10/12 03:54:35 grouping operations into packages
2016/10/12 03:54:35 planning meta data and facades
2016/10/12 03:54:35 rendering 1 models
2016/10/12 03:54:35 rendered model template: IP
2016/10/12 03:54:35 Error generating models: error generating outputs: Use the pre-defined default 400 response 'DefaultBadRequest' instead of defining your own
```

so I followed the instructions... ![](https://psr.slack.com/emoji/chompy/3f6fe5cb31c69b3d.gif)

I would also say if these are offlimits they ideally would be internal like you said - they come up in my code completion when I'm looking at `models` and so I thought they were fair game.